src/ScheduleFunctions.cpp

```diff
@@ -206,14 +206,19 @@ Stmt build_loop_nest(
         }
     }
 
-    // Sort the predicate guards for the fused loops so they are as far outwards
-    // as possible. IfInnner should not be reordered to outside of a for loop.
-    for (int i = (int)nest.size() - n_predicates_inner - n_predicates; i < (int)nest.size() - n_predicates; i++) {
+    // Sort the ifs so they are as far outwards as possible.
+    // BoxesTouched trims the domain of a variable within a scope of if-then-else
+    // based on the likely condition. However, it doesn't do it transitively; it
+    // doesn't trim the domains of other variables that directly/indirectly
+    // depend on the original variable in the likely condition outside the scope
```

Comment:
I believe it *does* do it transitively. There's a lot of logic to handle trimming bounds of dependent vars. It's certainly harder for it though.

Reply:
Oh you just moved this comment. It must have been stale.